### PR TITLE
Do not use SQL trigonometric functions

### PR DESCRIPTION
**Pull request by lpsinger**
Not all SQL servers support `cos` and `sin`. SQLite, for example, lacks these functions.

Evaluating theses special functions on the Python side is a pattern that we will need to use for the HEALPix forward and reverse transformations.

### CODE REVIEW

File: healpix_alchemy/point.py [hunk 1]
```diff
@@ -1,25 +1,57 @@
...
+
+
 class Point:
     """Mixin class to add a point to a an SQLAlchemy declarative model."""
 
     def __init__(self, *args, ra=None, dec=None, **kwargs):
         super().__init__(*args, **kwargs)
         self.ra = ra
         self.dec = dec
+        self.x = get_x_for_ra_dec(ra, dec)
```
Comment (dannygoldstein) [inline healpix_alchemy/point.py:46]: Adding these three columns increases the size of the table by >2x, perhaps the backend should just store xyz and ra, dec should be implemented as python only properties?

Storing both xyz and ra,dec seems like overkill - shouldn't be necessary to store both.

**Comment by dannygoldstein** [comment]
Merge conflicts

File: healpix_alchemy/point.py (hunk 1 above)
Comment (lpsinger) [reply healpix_alchemy/point.py:48]: It don't think it actually increases the table size because those values were calculate for the index anyway.

If we remove the ra and dec columns, then we may have to calculate them on the fly when displaying these values.

Comment (dannygoldstein) [reply healpix_alchemy/point.py:48]: Any way to test this? Can we measure the size of a table + index in both scenarios?

File: healpix_alchemy/point.py [hunk 2]
```diff
@@ -1,26 +1,59 @@
...
+
+def get_y_for_ra_dec(ra, dec):
+    return np.sin(np.deg2rad(ra)) * np.cos(np.deg2rad(dec))
+
+
+def get_z_for_ra_dec(ra, dec):
+    return np.sin(np.deg2rad(dec))
+
+
+def get_x_for_context(context):
```
Comment (dannygoldstein) [inline healpix_alchemy/point.py:26]: This and the two subsequent functions could potentially be simplified with a factory pattern:

```
def context_factory(variable_name):
    def get_variable_for_context(context):
        params = context.get_current_parameters()
        func = __module__.dict[f'get_{variable_name}_for_ra_dec']
        return func(params['ra'], params['dec'])
    return get_variable_for_context
```
```

Comment (dannygoldstein) [reply healpix_alchemy/point.py:26]: Sorry for late re-review

**Comment by lpsinger** [comment]
@mcoughlin, would you please review this?

File: healpix_alchemy/point.py [hunk 3]
```diff
@@ -26,9 +26,24 @@ def __init__(self, *args, ra=None, dec=None, **kwargs):
     ra = Column(Float)
     dec = Column(Float)
 
+    x = Column(
```
Comment (mcoughlin) [inline healpix_alchemy/point.py:29]: Maybe somehow indicate in the doc that these do not increase the DB size?

File: healpix_alchemy/point.py [hunk 4]
```diff
@@ -61,16 +74,10 @@ def within(self, other, radius):
         bool
 
         """
-        sin_radius = sind(radius)
-        cos_radius = cosd(radius)
+        sin_radius = sin(rad(radius))
+        cos_radius = cos(rad(radius))
         carts = (obj.cartesian for obj in (self, other))
-        terms = ((lhs.between(rhs - 2 * sin_radius, rhs + 2 * sin_radius),
+        terms = ((between(lhs, rhs - 2 * sin_radius, rhs + 2 * sin_radius),
```
Comment (mcoughlin) [inline healpix_alchemy/point.py:80]: Not part of this PR really, but maybe some pointer to the expression sources?

**Review by mcoughlin — APPROVED** [review]
Nice to remove the superfluous math.py. A few suggestions for comments otherwise nice.

File: healpix_alchemy/point.py (hunk 2 above)
Comment (lpsinger) [reply healpix_alchemy/point.py]: I did something similar.

File: healpix_alchemy/point.py (hunk 4 above)
Comment (lpsinger) [reply healpix_alchemy/point.py:78]: Isn't it clear what it's doing from the name of the `bounding_box_terms` variable?

Comment (mcoughlin) [reply healpix_alchemy/point.py:78]: Sure. Just might deserve a comment, especially as that line is just "terms"

Comment (lpsinger) [reply healpix_alchemy/point.py:78]: Done, see 0c395b9e14c1bceb714e2e27a8a28325b0bd18a4.

Comment (mcoughlin) [reply healpix_alchemy/point.py:80]: Looks good to me.

File: healpix_alchemy/point.py (hunk 3 above)
Comment (lpsinger) [reply healpix_alchemy/point.py:30]: I didn't actually measure the database size. We have the times of unit tests reported; it would be great to also report the size of the database. What's the easiest way to get the database size in SQLAlchemy or PostgreSQL?

Comment (mcoughlin) [reply healpix_alchemy/point.py:30]: Hmm looks like calls to postgresql:
https://writeonly.wordpress.com/2010/06/11/snippet-size-of-postgresql-tables-using-sqlalchemy/

Comment (lpsinger) [reply healpix_alchemy/point.py:30]: Also https://wiki.postgresql.org/wiki/Disk_Usage

File: healpix_alchemy/point.py (hunk 1 above)
Comment (lpsinger) [reply healpix_alchemy/point.py]: The unit test report now includes total tables + indices size for each unit test. The impact of this patch is very small.

## On main branch

```
================================ database size =================================
healpix_alchemy/tests/test_point.py::test_cross_join[10]: 952 kB
healpix_alchemy/tests/test_point.py::test_cross_join[100]: 952 kB
healpix_alchemy/tests/test_point.py::test_cross_join[1000]: 1304 kB
healpix_alchemy/tests/test_point.py::test_cross_join[10000]: 4232 kB
healpix_alchemy/tests/test_point.py::test_self_join[10]: 952 kB
healpix_alchemy/tests/test_point.py::test_self_join[100]: 952 kB
healpix_alchemy/tests/test_point.py::test_self_join[1000]: 1304 kB
healpix_alchemy/tests/test_point.py::test_self_join[10000]: 4232 kB
healpix_alchemy/tests/test_point.py::test_cone_search[10]: 952 kB
healpix_alchemy/tests/test_point.py::test_cone_search[100]: 952 kB
healpix_alchemy/tests/test_point.py::test_cone_search[1000]: 1304 kB
healpix_alchemy/tests/test_point.py::test_cone_search[10000]: 4232 kB
healpix_alchemy/tests/test_point.py::test_cone_search_literal_lhs[10]: 952 kB
healpix_alchemy/tests/test_point.py::test_cone_search_literal_lhs[100]: 952 kB
healpix_alchemy/tests/test_point.py::test_cone_search_literal_lhs[1000]: 1304 kB
healpix_alchemy/tests/test_point.py::test_cone_search_literal_lhs[10000]: 4232 kB
healpix_alchemy/tests/test_point.py::test_cone_search_literal_rhs[10]: 952 kB
healpix_alchemy/tests/test_point.py::test_cone_search_literal_rhs[100]: 952 kB
healpix_alchemy/tests/test_point.py::test_cone_search_literal_rhs[1000]: 1304 kB
healpix_alchemy/tests/test_point.py::test_cone_search_literal_rhs[10000]: 4232 kB
healpix_alchemy/tests/test_regions.py::test_fields: 1432 kB
```

## With this PR

```
================================ database size =================================
healpix_alchemy/tests/test_point.py::test_cross_join[10]: 952 kB
healpix_alchemy/tests/test_point.py::test_cross_join[100]: 968 kB
healpix_alchemy/tests/test_point.py::test_cross_join[1000]: 1352 kB
healpix_alchemy/tests/test_point.py::test_cross_join[10000]: 4712 kB
healpix_alchemy/tests/test_point.py::test_self_join[10]: 952 kB
healpix_alchemy/tests/test_point.py::test_self_join[100]: 968 kB
healpix_alchemy/tests/test_point.py::test_self_join[1000]: 1352 kB
healpix_alchemy/tests/test_point.py::test_self_join[10000]: 4712 kB
healpix_alchemy/tests/test_point.py::test_cone_search[10]: 952 kB
healpix_alchemy/tests/test_point.py::test_cone_search[100]: 968 kB
healpix_alchemy/tests/test_point.py::test_cone_search[1000]: 1352 kB
healpix_alchemy/tests/test_point.py::test_cone_search[10000]: 4712 kB
healpix_alchemy/tests/test_point.py::test_cone_search_literal_lhs[10]: 952 kB
healpix_alchemy/tests/test_point.py::test_cone_search_literal_lhs[100]: 968 kB
healpix_alchemy/tests/test_point.py::test_cone_search_literal_lhs[1000]: 1352 kB
healpix_alchemy/tests/test_point.py::test_cone_search_literal_lhs[10000]: 4712 kB
healpix_alchemy/tests/test_point.py::test_cone_search_literal_rhs[10]: 952 kB
healpix_alchemy/tests/test_point.py::test_cone_search_literal_rhs[100]: 968 kB
healpix_alchemy/tests/test_point.py::test_cone_search_literal_rhs[1000]: 1352 kB
healpix_alchemy/tests/test_point.py::test_cone_search_literal_rhs[10000]: 4712 kB
healpix_alchemy/tests/test_regions.py::test_fields: 1432 kB
```

File: healpix_alchemy/point.py (hunk 3 above)
Comment (lpsinger) [reply healpix_alchemy/point.py:30]: Well, adding indices _does_ increase the table size.
